### DISCUSSION TITLE
Is the documentation regarding the React Native `use_frameworks!` compatibility still up to date?

### DISCUSSION THREAD

**Discussion by Yonom** [invertase/react-native-firebase#7579] [General]
RN-Firebase's docs state that `use_frameworks!` is not compatible with Flipper & Fabric.

# Flipper

As per https://docs.expo.dev/guides/using-flipper/, Flipper is no longer recommended:

> React Native no longer recommends Flipper ([RFC0641](https://github.com/react-native-community/discussions-and-proposals/blob/main/proposals/0641-decoupling-flipper-from-react-native-core.md)). Features from Flipper, such as React Developer Tools and the network inspector, are available directly through Expo CLI in SDK 49 and greater.

I find the currently prominent warning on the rn-firebase website too scary

# New Architecture / Fabric

While researching the topic, I found a working group deep dive mentioning some fixes to Fabric that were planned for RN 0.72:
https://github.com/reactwg/react-native-new-architecture/discussions/115

I couldn't find a confirmation that the fix actually landed yet. I'm also completely clueless on how to test for this.

Is the documentation regarding the React Native `use_frameworks!` compatibility with the New Architecture / Fabric still up to date?

**Comment by mikehardy** [comment]
Flipper is no longer recommended for react-native as I understand it, so there will be no fixes forthcoming.

It appears based on other reports that in react-native 0.72.x new architecture will work, but not in bridgeless mode

So I *believe* that with the new adapter / transition layer that react-native put in 0.72 new architecture can work. I have not personally tested it, but if you do and want to make a PR to the docs noting for react-native 0.72+ in transition / bridged mode it should work, that would be fantastic

**Reply by DeveloperMCD** [reply]
I can't compile on iOS, though everything works great on Android.
The error I'm getting is with the library "React Native Vision Camera"

However, the SAME APP compiles fine on Android. Vision Camera, Firebase, and all. 

It might be because of the tweaks to my Podfile that were required, per the documentation:

```
platform :ios, min_ios_version_supported
prepare_react_native_project!
#use_modular_headers!
use_frameworks! :linkage => :static
$RNFirebaseAsStaticFramework = true
# Override Firebase SDK Version
$FirebaseSDKVersion = '10.20.0'

setup_permissions([
  'AppTrackingTransparency',
  'Camera',
  'LocationAccuracy',
  'LocationAlways',
  'LocationWhenInUse',
  'MediaLibrary',
  'PhotoLibrary'
])

# If you are using a `react-native-flipper` your iOS build will fail when `NO_FLIPPER=1` is set.
# because `react-native-flipper` depends on (FlipperKit,...) that will be excluded
#
# To fix this you can also exclude `react-native-flipper` using a `react-native.config.js`
# ```js
# module.exports = {
#   dependencies: {
#     ...(process.env.NO_FLIPPER ? { 'react-native-flipper': { platforms: { ios: null } } } : {}),
# ```
flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled

linkage = ENV['USE_FRAMEWORKS']
if linkage != nil
  Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
  use_frameworks! :linkage => linkage.to_sym
end

target 'stackbay' do
  config = use_native_modules!
  
  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    :hermes_enabled => flags[:hermes_enabled],
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    #:flipper_configuration => flipper_config,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  target 'stackbayTests' do
    inherit! :complete
    # Pods for testing
  end

  post_install do |installer|
    # https://github.com/facebook/react-native/blob/main/packages/react-native/scripts/react_native_pods.rb#L197-L202
    react_native_post_install(
      installer,
      config[:reactNativePath],
      :mac_catalyst_enabled => false
    )
    __apply_Xcode_12_5_M1_post_install_workaround(installer)

    installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        if Gem::Version.new('13.0') > Gem::Version.new(config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'])
          config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'
        end
      end
    end

    installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        if config.build_settings['WRAPPER_EXTENSION'] == 'bundle'
          config.build_settings['DEVELOPMENT_TEAM'] = '662Y85Q9D6'
        end
  
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.4'
  
        # **************Just add this line in your Podfile**************
        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', '_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION']
      end
    end    
  
  end
end
```

**Reply by mikehardy** [reply]
So...The error I'm getting is with the library "React Native Vision Camera"...sounds like react-native-vision-camera may not be compatible with `use_frameworks` style compilation? You should ask in their repo
